frontend/app/src/store/history/index.ts

```diff
@@ -1,17 +1,916 @@
+    ignore: boolean
+  ) => {
+    try {
+      ignore
+        ? await api.ignoreActions(actionIds, type)
+        : await api.unignoreActions(actionIds, type);
+      await fetchIgnored();
+    } catch (e: any) {
+      let title: string;
+      let description: string;
+      if (ignore) {
+        title = i18n.t('actions.ignore.error.title').toString();
+      } else {
+        title = i18n.t('actions.unignore.error.title').toString();
+      }
+
+      if (ignore) {
+        description = i18n
+          .t('actions.ignore.error.description', { error: e.message })
+          .toString();
+      } else {
+        description = i18n
+          .t('actions.unignore.error.description', { error: e.message })
+          .toString();
+      }
+      store.commit(
+        'setMessage',
+        {
+          success: false,
+          title,
+          description
+        } as Message,
+        { root: true }
+      );
+      return { success: false };
+    }
+
+    return { success: true };
+  };
+
+  // Purge
+  const purgeHistoryLocation = async (exchange: SupportedExchange) => {
+    fetchTrades(true, exchange).then();
+    fetchAssetMovements(true, exchange).then();
+    fetchLedgerActions(true, exchange).then();
+  };
+
+  const purgeExchange = async (
+    exchange: SupportedExchange | typeof ALL_CENTRALIZED_EXCHANGES
+  ) => {
+    function resetStatus(section: Section) {
+      setStatus(Status.NONE, section, store.getters['status'], store.commit);
+    }
+    if (exchange === ALL_CENTRALIZED_EXCHANGES) {
+      resetStatus(Section.TRADES);
+      resetStatus(Section.ASSET_MOVEMENT);
+      resetStatus(Section.LEDGER_ACTIONS);
+    } else {
+      await purgeHistoryLocation(exchange);
+    }
+  };
+
+  // Gitcoin
+  const fetchGitcoinGrant = async (
+    payload: GitcoinGrantEventsPayload
+  ): Promise<ActionResult<GitcoinGrants>> => {
+    try {
+      const { awaitTask } = useTasks();
+      const { taskId } = await api.history.gatherGitcoinGrandEvents(payload);
+      const meta: TaskMeta = {
+        title: i18n
+          .t('actions.balances.gitcoin_grant.task.title', {
+            grant: 'grantId' in payload ? payload.grantId : ''
+          })
+          .toString(),
+        numericKeys: balanceKeys
+      };
+      const { result } = await awaitTask<GitcoinGrants, TaskMeta>(
+        taskId,
+        TaskType.GITCOIN_GRANT_EVENTS,
+        meta
+      );
+      return { result, message: '' };
+    } catch (e: any) {
+      return {
+        result: {},
+        message: e.message
+      };
+    }
+  };
+
+  // Reset
+  const reset = () => {
+    associatedLocations.value = [];
+    trades.value = defaultHistoricState<TradeEntry>();
+    assetMovements.value = defaultHistoricState<AssetMovementEntry>();
+    transactions.value = defaultHistoricState<EthTransactionEntry>();
+    ledgerActions.value = defaultHistoricState<LedgerActionEntry>();
+    ignored.value = {};
+  };
+
+  return {
+    // Associated Locations
```

Comment:
The comments for grouping are kind of redundant (plus I am not the biggest fun of using comments to categorize sections), you can still get where everything belongs by the name. Maybe instead of that we could make `setupX` functions for each subcategory to keep everything self-contained. What do you think about using such an approach?

Reply:
I think so, I thought we still need to keep it in same defineStore. I will separate each of them